src/condor_utils/file_transfer.cpp

```diff
@@ -1887,10 +1887,24 @@ FileTransfer::ReadTransferPipeMsg()
 								   sizeof( int ) );
 		if(n != sizeof( int )) goto read_failed;
 
+		int stats_len = 0;
 		n = daemonCore->Read_Pipe( TransferPipe[0],
-								   (char *)&Info.num_cedar_files,
+								   (char *)&stats_len,
 								   sizeof( int ) );
 		if(n != sizeof( int )) goto read_failed;
+		if (stats_len) {
+			char *stats_buf = new char[stats_len];
+			n = daemonCore->Read_Pipe( TransferPipe[0],
+									stats_buf,
+									stats_len );
+			if(n != stats_len) {
+				delete [] stats_buf;
+				goto read_failed;
+			}
+			stats_buf[stats_len] = '\0';
+			classad::ClassAdParser parser;
+			parser.ParseClassAd(stats_buf, Info.stats);
+		}
```

Comment:
Don't leak the buffer.
```suggestion
			delete [] stats_buf;
		}
```